stdlib/public/runtime/Casting.cpp

```diff
@@ -2755,6 +2755,51 @@ swift::swift_conformsToProtocol(const Metadata *type,
   goto recur;
 }
 
+/// Return the type metadata for a given type name.
+///
+/// This is a temporary implementation to support some Foundation use cases;
+/// it will only lookup a type that directly implements a protocol as only
+/// those types will have conformance table entries.
+static const Metadata *
+_metadataForName(const llvm::StringRef typeName) {
+  installCallbacksToInspectDylib();
```

Comment:
In d0be84e1f3ac1b979583fb54786bdb82b6074b31 I refactored the conformance cache initialization to include installing the callbacks without deadlocking (that I could see).